src/infrastructure/events/OnCommentAdded.ts

```diff
@@ -0,0 +1,47 @@
+import { ApolloClient } from 'apollo-boost'
+
+import { AppEventListener } from '../../domain/events/Dispatcher'
+
+import { queries } from '../repository/SpotRepositoryGql'
+import {
+  FetchSpot,
+  FetchSpotVariables,
+} from '../repository/queries/__generated__/FetchSpot'
+
+export interface Dependencies {
+  apollo: ApolloClient<any>
+}
+
+export const OnCommentAdded = ({
+  apollo,
+}: Dependencies): AppEventListener => event => {
+  if (event.type === 'comment_added') {
+    const data = apollo.cache.readQuery<FetchSpot, FetchSpotVariables>({
+      query: queries.SPOT,
+      variables: {
+        id: event.comment.spotId,
+      },
+    })
+
+    if (!data) {
+      return
+    }
+
+    data.spot.comments.push({
```

Comment:
How difficult will it be to stay in sync with the actual app state coming from the future database?

How can we make sure that a data structure change doesn't break this? Reading this, it looks like at this step we need to know the GraphQL structure of a spot comment. This seems quite cumbersome at first glance.

Reply:
When the schema changes, the generated types will also change. Even though typings are not explicit in this code, every thing is strongly typed so we can rely on the compiler for adding things to the cache :)